lpc55_sign/src/signed_image.rs

```diff
@@ -259,8 +264,12 @@ pub fn sign_image(
 
     let (_, root0) = x509_parser::parse_x509_certificate(&root0_bytes)?;
 
-    let root0_pubkey =
-        rsa::RsaPublicKey::from_pkcs1_der(root0.public_key().subject_public_key.as_ref())?;
+    let root0_pubkey = rsa::RsaPublicKey::from_pkcs1_der(
+        root0.public_key().subject_public_key.as_ref(),
+    )
+    .or_else(|_| {
+        rsa::RsaPublicKey::from_public_key_der(root0.public_key().subject_public_key.as_ref())
```

Comment:
I don't think this is right.  `root0.public_key()` returns a SubjectPublicKeyInfo which is described in https://datatracker.ietf.org/doc/html/rfc5280#section-4-1-2-7.  The contents of `subject_public_key` depend on the algorithm ID.  For RSA, that's documented in https://datatracker.ietf.org/doc/html/rfc3279#section-2-3-1 and should be 1.2.840.113549.1.1 which says `subject_public_key` must contains a PKCS#1 RSAPublicKey.  AFAICT, that's what pki-playground is generating.  There should never be a PKCS#8 public key there unless a different algorithm ID is used.